lib/Auth/AuthHandler.php

```diff
@@ -0,0 +1,30 @@
+<?php
+namespace Spark\Auth;
+
+use Auryn\Injector;
+use Psr\Http\Message\ResponseInterface as Response;
+use Psr\Http\Message\ServerRequestInterface as Request;
+
+class AuthHandler
+{
+
+    protected $authenticator;
+    protected $injector;
+
+    public function __construct(AbstractAuthenticator $authenticator)
+    {
+        $this->authenticator = $authenticator;
+    }
+
+    public function __invoke(Request $request, Response $response, callable $next)
+    {
+
+        $token = $this->authenticator->getToken($request);
+
+        if ($token && !$this->authenticator->isValid($token)) {
+            throw new \Exception($this->authenticator->getErrorMessage());
```

Comment:
Do you think the token should be passed into the domain, or stored on the `Authenticator` class? I'm leaning towards storing it in the class.